longhaul/longhaul.md

```diff
@@ -0,0 +1,235 @@
+# IoT SDK Long Haul Technical Spec
+
+The Longhaul tests are to ensure the SDK can handle the rigors of in-field life.
+
+## Delivery scheduled
+
+* Longhaul tests will follow a crawl-\>walk-\>run approach.  
+
+* onghaul tests will sim-ship tests for both node and python using the Horton framework.
+
+## Parameters for Longhaul Testing
+
+* The test should run between 12 and 336 hours (2 weeks), but could be run for much longer periods
+
+* The test will perform a variety of operations and verify the results.  For example, D2C telemetry test will verify that the telemetry is sent successfully and also  arrived at the EventHub
+
+### Crawl
+
+* Tests will send D2C messages at a fixed cadence and verify receipt
+
+* Tests will use those same D2C to deliver time-series telemetry (such as memory use, ) to the hub for dashboarding
+
+* Tests will use reported properties to report test configuration and current status to the hub (elapsed time, average latency, etc)
+
+* Tests will run until failure, 
```

Comment:
This is one of those thing that everyone debates.  Should we run until time expires and then accumulate the errors or just stop at the first error.

Reply:
if tied to releases, then we should definitely be alerted at first failure although it could prove valuable to see what additional errors/delayed recovery might occur after the first error

Reply:
adding # of allowed failures as a config property along with slowness threshold and # allowed slow ops

---
In reply to: [441139944](https://github.com/Azure/azure-iot-sdk-python/pull/595#discussion_r441139944) [](ancestors = 441139944)

Reply:
made this configurable.

---
In reply to: [442348923](https://github.com/Azure/azure-iot-sdk-python/pull/595#discussion_r442348923) [](ancestors = 442348923)